interface-definitions/containers.xml.in

```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0"?>
+<interfaceDefinition>
+  <node name="container" owner="${vyos_conf_scripts_dir}/containers.py">
+    <properties>
+      <help>Container applications</help>
+    </properties>
+    <children>
+      <tagNode name="name">
+        <properties>
+          <help>Container name</help>
+        </properties>
+        <children>
+          <leafNode name="allow-host-networks">
+            <properties>
+              <help>Allow host networks in container</help>
+              <valueless/>
+            </properties>
+          </leafNode>
+          <leafNode name="description">
+            <properties>
+              <help>Container description</help>
+            </properties>
+          </leafNode>
+          <leafNode name="image">
+            <properties>
+              <help>Image name in the hub-registry</help>
+            </properties>
+          </leafNode>
+          <tagNode name="network">
+            <properties>
+              <help>Attach user defined network to container</help>
+              <completionHelp>
+                <path>container network</path>
+              </completionHelp>
+            </properties>
+            <children>
+              <leafNode name="address">
+                <properties>
+                  <help>Set IPv4 static address to container (optional)</help>
+                  <valueHelp>
+                    <format>ipv4</format> 
+                    <description>IPv4 address (x.x.x.1 reserved)</description>
+                  </valueHelp>
+                  <constraint>
+                    <validator name="ipv4-address"/>
+                  </constraint>
+                </properties>
+              </leafNode>
+            </children>
+          </tagNode>
+        </children>
+      </tagNode>
+      <tagNode name="network">
+        <properties>
+          <help>Network name</help>
+        </properties>
+        <children>
+          <leafNode name="description">
+            <properties>
+              <help>Network description</help>
+            </properties>
+          </leafNode>
+          <leafNode name="ipv4-prefix">
```

Comment:
Is there a reason to explicitly name this ipv4? Also the validator you use below support both ipv4 and ipv6

Reply:
> Is there a reason to explicitly name this ipv4? Also the validator you use below support both ipv4 and ipv6

Explicitly name needs for adding many containers to one user-defined network.
So we need unique value for user-defined networks.
```
set container name alp101 image 'alpine'
set container name alp101 network NET55
set container name alp102 image 'alpine'
set container name alp102 network NET55
set container network NET55 ipv4-prefix '10.5.55.0/24'

```
Attach to alp101 and ping alp102
```
vyos@r4-roll# sudo podman exec -it alp101 /bin/sh

/ # ip a show dev eth0
3: eth0@if7: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue state UP 
    link/ether 1a:6b:aa:f0:d9:61 brd ff:ff:ff:ff:ff:ff
    inet 10.5.55.2/24 brd 10.5.55.255 scope global eth0

/ # ping 10.5.55.3
PING 10.5.55.3 (10.5.55.3): 56 data bytes
64 bytes from 10.5.55.3: seq=0 ttl=42 time=0.115 ms
64 bytes from 10.5.55.3: seq=1 ttl=42 time=0.183 ms

```
If we need to attach containers to different user-defined networks we need some unique values for it.
As we starting with only ipv4-prefix right now, we use that validator.